docs/variants.md

```diff
@@ -154,3 +154,93 @@ requirements:
   run:
   - libcurl >=8,<9
 ```
+
+## Prioritizing variants
+
+You might produce multiple variants for a package, but want to define a _priority_ for a given variant.
+The variant with the highest priority would be the default package that is selected by the resolver.
+
+There are two mechanisms to make this possible: `mutex` packages and the `down_prioritize_variant` option in the recipe.
+
+### The "down_prioritize_variant" option
+
+!!! note
+    It is not always necessary to use the `down_prioritize_variant` option - only if the solver has no other way to
+    prefer a given variant. For example, if you have a package that has multiple variants for different Python versions,
+    the solver will automatically prefer the variant with the highest Python version.
+
+The `down_prioritize_variant` option allows you to specify a variant that should be _down-prioritized_. For example:
+
+```yaml title="recipe.yaml" hl_lines="7"
+build:
+  variant_config:
+    use_keys:
+      # use cuda from the variant config, e.g. to build multiple CUDA variants
+      - cuda
+    # this will down-prioritize the cuda variant versus other variants of the package
+    down_prioritize_variant: ${{ -1 if cuda else 0 }}
+```
+
+### Mutex packages
+
+Another way to make sure the right variants are selected are "mutex" packages. A mutex package is a package that is
+mutually exclusive. We use the fact that only one package of a given name can be installed at a time (the solver has to choose).
+
+A mutex package might be useful to make sure that all packages that depend on BLAS are compiled against the same BLAS implementation.
+The mutex package will serve the purpose that "openblas" and "mkl" can never be installed at the same time.
+
+We could define a BLAS mutex package like this:
+
+```yaml title="variant_config.yaml"
+blas_variant:
+  - "openblas"
+  - "mkl"
+```
+
+And then the `recipe.yaml` for the `mutex` package could look like this:
+
+```yaml title="recipe.yaml" hl_lines="6 9"
+package:
+  name: blas_mutex
+  version: 1.0
+
+build:
+  string: ${{ blas_variant }}
```

Comment:
How about using `${{ blas_variant }}${{ hash }}_${{ build_number }}` to not encourage users to create packages without hashes in the build string?